ocs_ci/utility/kms.py

```diff
@@ -1413,7 +1416,332 @@ def validate_external_hpcs(self):
+            )
+            key_ids.append(base64.b64decode(run_cmd(cmd=cmd)).decode())
+        return key_ids
+
+    def get_noobaa_key_id(self):
+        """
+        Retrieve the key ID used for encryption by Noobaa
+
+        Returns:
+            (str): Key ID used by NooBaa for encryption
+
+        """
+        cmd = (
+            rf"oc get cm ocs-kms-connection-details -o jsonpath=\"{{.data[\'KMIP_SECRET_NAME\']}}\""
+            f" -n {constants.OPENSHIFT_STORAGE_NAMESPACE}"
+        )
+        secret_name = run_cmd(cmd=cmd)
+
+        cmd = (
+            rf"oc get secret {secret_name} -o jsonpath=\"{{.data[\'UniqueIdentifier\']}}\""
+            f" -n {constants.OPENSHIFT_STORAGE_NAMESPACE}"
+        )
+        noobaa_key_id = base64.b64decode(run_cmd(cmd=cmd)).decode()
+        return noobaa_key_id
+
+    def post_deploy_verification(self):
+        """
+        Verify ODF deployment using KMIP
+
+        """
+        self.validate_ciphertrust_deployment()
+
+    def validate_ciphertrust_deployment(self):
+        """
+        Verify whether OSD and NooBaa keys are stored in CipherTrust Manager
+
+        """
+
+        key_id_list = self.get_key_list_ciphertrust()
+
+        # Check for OSD keys
+        osd_key_ids = self.get_osd_key_ids()
+        if all(id in key_id_list for id in osd_key_ids):
+            logger.info("KMIP: All OSD keys found in CipherTrust Manager")
+        else:
+            logger.error(
+                "KMIP: Only some or no OSD keys found in CipherTrust Manager"
+                f"Keys found in CipherTrust Manager: {key_id_list}"
+                f"OSD keys from the ODF cluster: {osd_key_ids}"
+            )
+            raise NotFoundError(
+                "KMIP: Only some or no OSD keys found in CipherTrust Manager"
+            )
+
+        # Check for NOOBAA key
+        noobaa_key_id = self.get_noobaa_key_id()
+        if noobaa_key_id in key_id_list:
+            logger.info("KMIP: Noobaa encryption key found in CipherTrust Manager")
+        else:
+            raise NotFoundError(
+                "KMIP: Noobaa encryption key found in CipherTrust Manager"
+            )
+
+        # Check kms enabled
+        if not is_kms_enabled():
+            logger.error("KMS not enabled on storage cluster")
+            raise NotFoundError("KMS flag not found")
+
+    def cleanup(self):
+        """
+        Cleanup for KMIP
+
+        """
+        self.update_kmip_env_vars()
+
+        # Retrieve OSD and NooBaa keys for deletion and delete
+        key_ids = self.get_osd_key_ids()
+        key_ids.append(self.get_noobaa_key_id())
+        if not key_ids:
+            logger.warning("No keys found to be deleted from CipherTrust")
+
+        else:
+            for key in key_ids:
+                self.delete_ciphertrust_key(key)
+            logger.info("Keys deleted from CipherTrust Manager")
+
+
+kms_map = {"vault": Vault, "hpcs": HPCS, "kmip": KMIP}
```

Comment:
Just noticing this now but do we want to move this towards the top of the module (under the imports), it seems a bit hidden down here.

Reply:
Moving it above, throws undefined errors

Reply:
Where did you try to move it to? I can't see why you'd hit undefined errors moving it towards the top of the file as long as it was in the same scope.

Reply:
I  moved it below the imports section, after line 53:
```
logger = logging.getLogger(__name__)

kms_map = {"vault": Vault, "hpcs": HPCS, "kmip": KMIP}
```

This is the error I see:
```
  File "/home/rgeorge/ocs-ci/ocs_ci/utility/kms.py", line 55, in <module>
    kms_map = {"vault": Vault, "hpcs": HPCS, "kmip": KMIP}
NameError: name 'Vault' is not defined

```

Reply:
Ah I understand now, no worries this is good here.